sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -131,26 +175,30 @@ public Mono<Void> updateCheckpoint(Checkpoint checkpoint) {
                 .propagate(new IllegalStateException(
                     "Checkpoint is either null, or both the offset and the sequence number are null.")));
         }
-        String prefix = prefixBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup());
-        String key = keyBuilder(prefix, checkpoint.getPartitionId());
+        byte[] prefix = prefixBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup());
```

Comment:
I would wrap this in a Mono so it's possible to run this asynchronously rather than synchronously.

```java
return Mono.fromRunnable(() -> {
   byte[] prefix...
   // rest of your stuff
});
```